packages/react/src/components/Calendar/CalendarDay/CalendarDay.base.tsx

```diff
@@ -168,7 +168,7 @@ const CalendarDayNavigationButtons = (props: ICalendarDayNavigationButtonsProps)
       {showCloseButton && (
         <button
           className={css(classNames.headerIconButton)}
-          onClick={onDismiss}
+          onClick={() => onDismiss?.()}
```

Comment:
I think the correct thing to do here would be to not make this change here, but instead move it up to `DatePicker.base.tsx` and do something similar when passing `calendarDismissed` to `onDismiss` in `CalendarType`.

Also, when you make that change, it'd be great if you could scaffold this into its own non-inlined function so that it doesn't create a new reference on every render.

Reply:
@khmakoto Thanks for reviewing. I have reverted the original change and applied the new change as you have requested